src/main/scala/org/apache/spark/shuffle/ucx/UcxWorkerWrapper.scala

```diff
@@ -188,7 +188,14 @@ case class UcxWorkerWrapper(worker: UcpWorker, transport: UcxShuffleTransport, i
   def connectByWorkerAddress(executorId: transport.ExecutorId, workerAddress: ByteBuffer): Unit = {
     logDebug(s"Worker $this connecting back to $executorId by worker address")
     val ep = worker.newEndpoint(new UcpEndpointParams().setName(s"Server connection to $executorId")
-      .setUcpAddress(workerAddress))
+      .setUcpAddress(workerAddress)
+      .setPeerErrorHandlingMode()
```

Comment:
@yosefe does peerErrorHandling mode works with connect by worker address?

Reply:
In UCX,
```
ucs_status_ptr_t ucp_ep_close_nbx(ucp_ep_h ep, const ucp_request_param_t *param)
{
    ucp_worker_h  worker = ep->worker;
    void          *request = NULL;
    ucp_request_t *close_req;

    if ((ucp_request_param_flags(param) & UCP_EP_CLOSE_FLAG_FORCE) &&
        (ucp_ep_config(ep)->key.err_mode != UCP_ERR_HANDLING_MODE_PEER)) {
        return UCS_STATUS_PTR(UCS_ERR_INVALID_PARAM);
    }
```
when force to close the ep, it will also check the flag `UCP_ERR_HANDLING_MODE_PEER` setting.

Reply:
@gleon99 @yosefe  Can you pls help to review and merge this PR? Thanks!